database/changesets/20210709_TerraDataRepoSink.xml

```diff
@@ -0,0 +1,71 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext
+                        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd
+                        http://www.liquibase.org/xml/ns/dbchangelog
+                        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.8.xsd">
+    <changeSet author="hornet@broadinstitute.org"
+               dbms="postgresql"
+               id="20210709"
+               logicalFilePath="20210709_TerraDataRepoSink.xml"
+               runInTransaction="false">
+        <comment>
+            A descriptor for the type of TDR job wfl started while ingesting
+            outputs
+        </comment>
+        <sql>
+            CREATE TYPE TDRJobType AS ENUM (
+                'FileLoad',
+                'MetadataIngest'
+            )
+        </sql>
+        <comment>
+            The status of the TDR job wfl started, as reported by TDR. See
+            https://data.terra.bio/swagger-ui.html#/repository/retrieveJob
+            for details.
+        </comment>
+        <sql>
+            CREATE TYPE TDRJobStatus AS ENUM (
+                'running',
+                'succeeded',
+                'failed'
+            )
+        </sql>
+        <comment>
+            The TDR Job Queue used by the `TerraDataRepo` Sink
+        </comment>
+        <sql>
+            CREATE TYPE TerraDataRepoSinkDetails AS (
+                id        BIGINT,       -- ALTER ADD GENERATED ALWAYS AS IDENTITY
```

Comment:
(For my own understanding) this comment is flagging the alteration we'll need to run following creation from the type to get an auto-incrementing id column.

Reply:
yeah exactly!